File: src/main/java/com/steelzack/xmladder/instruction/XmlAdderAddAttributeManager.java
package com.steelzack.xmladder.instruction;

import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;

/**
 * Created by joaofilipesabinoesperancinha on 18-02-16.
 */
public class XmlAdderAddAttributeManager {

    private Map<String, XmlAdderInstruction> xmlAdderInstructionArrayMap = new HashMap<>();

    public void addAddAttribute(String name, String value, String xpath) {
        XmlAdderInstruction instruction = xmlAdderInstructionArrayMap.get(xpath);
        if (instruction == null) {
            instruction = new XmlAdderInstruction(null, new HashMap<>(), null, new HashSet<>(), null, null);
            instruction.addAddAttribute(name, value);
            xmlAdderInstructionArrayMap.put(xpath, instruction);
        } else {
            instruction.addAddAttribute(name, value);
        }
    }

    public void addRemoveAttribute(String name, String xpath) {
        XmlAdderInstruction instruction = xmlAdderInstructionArrayMap.get(xpath);
        if (instruction == null) {
            instruction = new XmlAdderInstruction(null, new HashMap<>(), null, new HashSet<>(), null, null);
            instruction.addDeleteAttribute(name);
            xmlAdderInstructionArrayMap.put(xpath, instruction);
        } else {
            instruction.addDeleteAttribute(name);
        }
    }

    public Map<String, XmlAdderInstruction> getXmlAdderInstructionArrayMap() {
        return xmlAdderInstructionArrayMap;
    }
}

File: jeorg-kotlin-apps/xml-adder/app/src/main/kotlin/org/jesperancinha/xml/adder/instruction/XmlAdderAddAttributeManager.kt
package org.jesperancinha.xml.adder.instruction

/**
 * Created by joaofilipesabinoesperancinha on 18-02-16.
 */
class XmlAdderAddAttributeManager {
    internal val xmlAdderInstructionArrayMap: MutableMap<String, XmlAdderInstruction> = HashMap()
    fun addAddAttribute(name: String?, value: String?, xpath: String?) = name?.let {
        var instruction = xmlAdderInstructionArrayMap[xpath]
        if (instruction == null) {
            instruction = XmlAdderInstruction(null, HashMap(), null, HashSet(), null, null)
            instruction.addAddAttribute(name, value)
            xpath?.let { xmlAdderInstructionArrayMap[xpath] = instruction }
        } else {
            instruction.addAddAttribute(name, value)
        }
    }

    fun addRemoveAttribute(name: String?, xpath: String?) {
        var instruction = xmlAdderInstructionArrayMap[xpath]
        if (instruction == null) {
            instruction = XmlAdderInstruction(null, HashMap(), null, HashSet(), null, null)
            instruction.addDeleteAttribute(name)
            xpath?.let { xmlAdderInstructionArrayMap[xpath] = instruction }
        } else {
            instruction.addDeleteAttribute(name)
        }
    }

    fun getXmlAdderInstructionArrayMap(): Map<String, XmlAdderInstruction> {
        return xmlAdderInstructionArrayMap
    }
}